docs/architecture/adr-023-data-fee-market.md

```diff
@@ -0,0 +1,60 @@
+# ADR 023: DA Fee Market
+
+## Status
+
+Proposed
+
+## Changelog
+
+- 2023/06/28: Initial draft
```

Comment:
[question] is this date correct? June 28th seems like a long time ago. Perhaps it meant to say October 27th? 

```suggestion
- 2023/10/27: Initial draft
```

Reply:
Lol I actually started writing this in June so it is somewhat correct but I'll update the date to now